concent_api/core/tests/test_integration_report_computed_task.py

```diff
@@ -2795,3 +2795,57 @@ def test_that_message_report_computed_task_in_database_is_replaced_when_requesto
         self._assert_stored_message_counter_not_increased()
 
         self._assert_client_count_is_equal(2)
+
+    def test_provider_forces_computed_task_with_non_positive_price_returns_service_refused(self):
+        """
+        Tests if on provider ForceReportComputedTask message Concent will return ServiceRefused
+        if TaskToCompute price value is non positive.
+
+        Expected message exchange:
+        Provider -> Concent:   ForceReportComputedTask
+        Concent  -> Provider:  ServiceRefused
+        """
+
+        # STEP 1: Provider forces computed task report via Concent
```

Comment:
Please, don't copy-paste this, reduce the setup to minimum and remove '#Step X:' comments if theye're not really needed.